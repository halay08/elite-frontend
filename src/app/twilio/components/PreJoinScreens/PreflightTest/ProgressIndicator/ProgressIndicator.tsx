import React from 'react';
import { makeStyles } from '@material-ui/core';
import { TEST_DURATION } from '../PreflightTest';

const SIZE = 24;
const THICKNESS = 3;
const OUTER_RADIUS = SIZE / 2;
const INNER_RADIUS = OUTER_RADIUS - THICKNESS;
const INNER_CIRCUMFERENCE = 2 * Math.PI * INNER_RADIUS;

const useStyles = makeStyles(() => ({
  svg: {
    transform: 'rotate(-90deg)',
    marginRight: '0.6em',
    fill: 'none',
  },
  innerCircle: {
    stroke: 'rgb(200, 204, 207)',
  },
  circle: {
    stroke: '#027AC5',
    strokeDasharray: INNER_CIRCUMFERENCE,
    strokeDashoffset: '0px',
    animation: `${TEST_DURATION / 1000 + 1}s $progress linear`,
  },
  '@keyframes progress': {
    '0%': {
      strokeDashoffset: `${INNER_CIRCUMFERENCE}px`,
    },
    '100%': {
      strokeDashoffset: '0px',
    },
  },
}));

export default function ProgressIndicator() {
  const classes = useStyles();

  return (
    <svg
      viewBox={`0 0 ${SIZE} ${SIZE}`}
      width={SIZE}
      height={SIZE}
      strokeWidth={`${THICKNESS}px`}
      xmlns="http://www.w3.org/2000/svg"
      className={classes.svg}
    >
      <circle
        cx={OUTER_RADIUS}
        cy={OUTER_RADIUS}
        r={INNER_RADIUS}
        className={classes.innerCircle}
      />
      <circle
        cx={OUTER_RADIUS}
        cy={OUTER_RADIUS}
        r={INNER_RADIUS}
        className={classes.circle}
      />
    </svg>
  );
}
