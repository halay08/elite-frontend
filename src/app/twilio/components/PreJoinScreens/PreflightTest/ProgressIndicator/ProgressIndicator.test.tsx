import React from 'react';
import renderer from 'react-test-renderer';
import ProgressIndicator from './ProgressIndicator';

describe('the ProgressIndicator component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<ProgressIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
