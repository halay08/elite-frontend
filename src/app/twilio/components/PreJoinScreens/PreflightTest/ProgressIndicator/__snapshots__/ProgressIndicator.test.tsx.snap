// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ProgressIndicator component should render correctly 1`] = `
<svg
  className="makeStyles-svg-1"
  height={24}
  strokeWidth="3px"
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className="makeStyles-innerCircle-2"
    cx={12}
    cy={12}
    r={9}
  />
  <circle
    className="makeStyles-circle-3"
    cx={12}
    cy={12}
    r={9}
  />
</svg>
`;
